packages/build-info/src/settings/get-toml-settings.ts

```diff
@@ -34,22 +26,60 @@ function gracefulParseToml<T>(content: string): T {
   }
 }
 
-export async function getNetlifyTomlSettings(fs: FileSystem, directory: string) {
+/** Mutates the provided settings by setting the property if the value is not undefined  */
+function addProperty<P extends keyof Settings>(
+  settings: Partial<Settings>,
+  property: P,
+  value?: Settings[P],
+): Partial<Settings> {
+  if (value) {
+    settings[property] = value
+  }
+  return settings
+}
+
+export async function getTomlSettingsFromPath(
+  fs: FileSystem,
+  directory: string,
+): Promise<Partial<Settings> | undefined> {
   const tomlFilePath = fs.join(directory, 'netlify.toml')
 
   try {
-    const settings: Partial<BuildSettings> = {}
-    const { build, functions, template } = gracefulParseToml<NetlifyTOML>(await fs.readFile(tomlFilePath))
+    const settings: Partial<Settings> = {}
+    const { build, dev, functions, template, plugins } = gracefulParseToml<NetlifyTOML>(await fs.readFile(tomlFilePath))
 
-    if (build) {
-      settings.cmd = build?.command
-      settings.dir = build.publish
-    }
+    addProperty(settings, 'buildCommand', build?.command)
```

Comment:
just a nit: I was thinking that this could reduce the cognitive complexity of having to understand the generics of `addProperty`, at the expense of making the code a little bit more verbose. but that might be worth it, especially for people onboarding into the codebase.

```suggestion
    settings.buildCommand = build?.command ?? settings.buildCommand
```

Reply:
Yea that's a good call :)